### ISSUE TITLE
Can Sketch render without anti-aliasing?

### ISSUE THREAD

**Issue by aykaramba** [vydd/sketch#172]
Can Sketch render without anti-aliasing to get the old school aliased / jagged edges look? I am fiddling with making Asteroids and other retro game like things, wondering if there is an setting that can control that.

**Comment by Kevinpgalligan** [comment]
This is more of an OpenGL question, I think, as that's the graphics API that Sketch uses under the hood. I tried investigating this by searching "OpenGL disable anti-aliasing", and it seems that OpenGL doesn't enable anti-aliasing by default?? Might be wrong though.

You could look into tweaking the OpenGL settings, maybe. Or something like this might work -- rendering to a lower-resolution buffer and then upscale it to fit the window.

**Comment by aykaramba** [comment]
Aha!  Okay, now I understand what to research.  Thank you very much!

**Comment by Kevinpgalligan** [comment]
It still might be a valid feature to add to Sketch, by the way! It's just that currently, Sketch doesn't offer a low-level API for graphics stuff.

**Comment by aykaramba** [comment]
Reopening this ticket for further exploration.

Thanks all for the feedback.  Gleefre, that does work as long as one doesn't use the (scale ... ) function. Screenshot:

<img width="430" height="464" alt="Image" src="https://github.com/user-attachments/assets/ec956436-6633-4048-89cc-f3a506c7511b" />

For the purposes of this conversation, this looks like a great way to hack a solution. Just thinking about this further, on the one hand it seems like my initial question is misplaced in a sense. On, the one hand, to achieve a pixel art style look in any other engine, you need to generate the appropriate sprites and then let the engine handle the rest. On the other, it would be really interesting to be able to flip Sketch into an 8bit style mode to resembling something like Tic80 or Pico8 projects.

QUESTION: Would anyone be able to comment on what a "proper" solution might require? How is this handled in something like Aesprite / Tic80 / Pico8? If it can be done, can it be done without losing the anti-aliasing?

Looking to understand this domain a little bit better.

Thanks.

**Comment by aykaramba** [comment]
So, thinking some more about this:

* We can turn off the anti-aliasing, even if it is a bit of a hack, it does work.
* When you turn off the anti-aliasing, it is difficult to see at normal resolution.
* What might be interesting would be if the rendering could be "resampled" at some arbitrary multiple of whatever equals to 1 pixel.

So instead of creating a 100x100 window, one might  be able to use some arbitrary window resolution but the rendering would "snap" to some multiple of 1 pixel and upscale the rendering without doing whatever (scale) does right now.  It would allow for arbitrarily large window sizes for really large monitors with pixelized rendering visible at arbitrary resolution.

Any comments and critiques welcome, just jotting down the thought for future review.

**Comment by aykaramba** [comment]
I couldn't resist. Over the weekend I prompted AI to generate some CL-SDL code and draw at low res then upscale to an arbitrary resolution without applying anti-aliasing to get that TIC80 fantasy computer style.  

I created an CL-SDL2 based demo of how to render at a lower resolution to a texture and render the texture to some arbitrary size window to get a TIC80 style pixel rendering without anti-aliasing.  Maybe this is one approach to add that style of rendering to Sketch.

Repo is here: https://github.com/aykaramba/tic80-pixel-art/tree/master
